### ISSUE TITLE
Identify S3 requests 

### ISSUE THREAD

**Issue by guy-har** [treeverse/lakeFS#2351]
lakeFS currently Identifies S3 requests (which should be handled by S3Gateway) by domain. This forces users to configure the S3 domain even in cases it is not needed. We would like to check if there is another way to identify if a request is an S3 request (for example, By checking for sigV4/sigV2 headers).

**Comment by guy-har** [comment]
We should take in consideration s3 [anonymous user access](https://docs.aws.amazon.com/AmazonS3/latest/userguide/example-bucket-policies.html#example-bucket-policies-use-case-2)
public buckets may be accessed without any signature
for example running 
```bash 
curl --location --head 'dev-lakefs-docs.s3.amazonaws.com/architecture.html'
```
would be a valid S3 request that is hard (maybe even impossible) to identify

**Comment by guy-har** [comment]
This issue was considered and solved in #2387
